### PR TITLE
Opengrok tools version --version

### PR DESCRIPTION
**Pull request by tulinkry**
Moving the version to a different python file so it's available when the commands are invoked with `--version`.

### CODE REVIEW

**Comment by tulinkry** [comment]
I will merge this after #2498 so the PATH in pom.xml is correct

File: opengrok-tools/src/main/python/opengrok_tools/config_merge.py [hunk 1]
```diff
@@ -25,7 +25,8 @@
 import logging
 import sys
 
-from .utils.java import Java, get_javaparser
+from .utils.java import Java
+from opengrok_tools.utils.parsers import get_javaparser
```
Comment (tulinkry) [inline opengrok-tools/src/main/python/opengrok_tools/config_merge.py]: Use relative import

File: opengrok-tools/src/main/python/opengrok_tools/indexer.py [hunk 2]
```diff
@@ -27,7 +27,7 @@
 import sys
 
 from .utils.indexer import FindCtags, Indexer
-from .utils.java import get_javaparser
+from opengrok_tools.utils.parsers import get_javaparser
```
Comment (tulinkry) [inline opengrok-tools/src/main/python/opengrok_tools/indexer.py]: Use relative import

File: opengrok-tools/src/main/python/opengrok_tools/java.py [hunk 3]
```diff
@@ -25,7 +25,8 @@
 import argparse
 import sys
 
-from .utils.java import Java, get_javaparser
+from .utils.java import Java
+from opengrok_tools.utils.parsers import get_javaparser
```
Comment (tulinkry) [inline opengrok-tools/src/main/python/opengrok_tools/java.py]: Use relative import

File: opengrok-tools/src/main/python/opengrok_tools/mirror.py [hunk 4]
```diff
@@ -36,18 +36,19 @@
 import sys
 import tempfile
 from logging.handlers import RotatingFileHandler
+
 from filelock import Timeout, FileLock
 
+from opengrok_tools.utils.parsers import get_baseparser
+from .scm.repository import RepositoryException
```
Comment (tulinkry) [inline opengrok-tools/src/main/python/opengrok_tools/mirror.py]: Use relative import

File: opengrok-tools/src/main/python/opengrok_tools/reindex_project.py [hunk 5]
```diff
@@ -28,7 +28,7 @@
 import tempfile
 
 from .utils.indexer import Indexer
-from .utils.java import get_javaparser
+from opengrok_tools.utils.parsers import get_javaparser
```
Comment (tulinkry) [inline opengrok-tools/src/main/python/opengrok_tools/reindex_project.py]: Use relative  import

File: opengrok-tools/src/main/python/opengrok_tools/utils/__init__.py [hunk 6]
```diff
@@ -15,5 +15,6 @@
     'hook',
     'repofactory',
     'webutil',
-    'readconfig'
+    'readconfig',
+    'parsers',
```
Comment (tulinkry) [inline opengrok-tools/src/main/python/opengrok_tools/utils/__init__.py]: Missing import

File: opengrok-tools/src/test/python/test_versions.py [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+            ('opengrok-reindex-project'),
+            ('opengrok-sync'),
+    ))
+    def test_opengrok_version(self, command):
+        """
+        Test that installed command has the version option
+        :param command: the command name
+        :return:
+        """
+        pass
```
Comment (vladak) [inline opengrok-tools/src/test/python/test_versions.py]: why the `pass` is needed here ?

File: opengrok-tools/src/test/python/test_versions.py [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+        cmd = Command([command, '--version'])
+        cmd.execute()
+        self.assertEqual(0, cmd.getretcode())
+        self.assertEqual(Command.FINISHED, cmd.getstate())
+        self.assertTrue(len(cmd.getoutputstr()) > 1)
```
Comment (vladak) [inline opengrok-tools/src/test/python/test_versions.py]: perhaps do a casual check for the version string contents ?

File: opengrok-tools/setup.py [hunk 9]
```diff
@@ -6,6 +6,28 @@
...
+def extract_version_from_path(*file_paths):
+    """
+    Look for a version in a file
+    see https://packaging.python.org/guides/single-sourcing-package-version/
+    :param file_paths: path to the file
+    :return: the version
+    :raises: an RuntimeError when version is not available
+    """
+
+    def read(*parts):
```
Comment (vladak) [inline opengrok-tools/setup.py]: use more descriptive name of the function

File: opengrok-tools/src/main/python/opengrok_tools/version.py [hunk 10]
```diff
@@ -0,0 +1,15 @@
+def get_version(version):
```
Comment (vladak) [inline opengrok-tools/src/main/python/opengrok_tools/version.py]: add a top-level comment that the file is being modified by the Maven build

File: opengrok-tools/pom.xml [hunk 11]
```diff
@@ -111,29 +111,41 @@ Portions Copyright (c) 2017-2018, Chris Fraire <cfraire@me.com>.
                         </configuration>
                     </execution>
                     <execution>
-                        <id>copy package resources</id>
+                        <id>copy version file</id>
```
Comment (vladak) [inline opengrok-tools/pom.xml]: and change the contents

Comment (tulinkry) [reply opengrok-tools/pom.xml]: done

File: opengrok-tools/src/main/python/opengrok_tools/version.py (hunk 10 above)
Comment (tulinkry) [reply opengrok-tools/src/main/python/opengrok_tools/version.py]: done

File: opengrok-tools/setup.py [hunk 12]
```diff
@@ -6,6 +6,28 @@
...
+        with open(os.path.join(SCRIPT_DIR, *parts), 'r') as fp:
+            version = {}
+            exec(fp.read(), version, None)
```
[the remark this answers is not in the document]
Comment (tulinkry) [reply opengrok-tools/setup.py]: import works

I tried that before but maybe I had some incorrect setup so it didn't work.

File: opengrok-tools/src/test/python/test_versions.py (hunk 8 above)
Comment (tulinkry) [reply opengrok-tools/src/test/python/test_versions.py]: changed

File: opengrok-tools/setup.py (hunk 9 above)
Comment (tulinkry) [reply opengrok-tools/setup.py]: deleted the whole function

**Comment by vladak** [comment]
need to fix the style issue and it's ready to go.

**Pull request merged**